lib/chef/resource/archive_file.rb

```diff
@@ -119,7 +123,8 @@ class ArchiveFile < Chef::Resource
           converge_by("set owner of files extracted in #{new_resource.destination} to #{new_resource.owner}:#{new_resource.group}") do
             archive = Archive::Reader.open_filename(new_resource.path)
             archive.each_entry do |e|
-              FileUtils.chown(new_resource.owner, new_resource.group, "#{new_resource.destination}/#{e.pathname}")
+              file_dest = strip_components_from_path(e.pathname)
```

Comment:
This seems to be repeated 3 times in this resource which is screaming for some sort of abstraction. Thinking a small class responsible for taking in an artifact and both stripping components and then returning a list of files that are applicable. Anyhow, I need to think about it more and I don't think it should hold this up given the unit test coverage but just wanted to dump what was in my brain.

Reply:
It looks to me like this functionality should go into `archive.each_entry(strip_components: new_resource.strip_components)` and the work should get done in ffi-libarchive and/or libarchive directly (no idea if libarchive supports that or not)

Reply:
Completely agree with that in a pure OO sense, but I didn't know if that was violating any other principles for FFI specifically. Are you good with a PR to extend functionality on top of libarchive?